public/javascripts/controllers/farmer-ctrl.js

```diff
@@ -129,19 +157,109 @@ angular.module('jasmic.controllers')
                 $scope.newFarm = !$scope.newFarm;
                 $scope.farm = {};
             };
+
+            $scope.editFarmLocation = function(arrayindex) {
+
+                $scope.editFarm = !$scope.editFarm;
+                $scope.farm = $scope.farmer.fr_farms[arrayindex];
+
+                $scope.editingFarm = arrayindex;//set the farm currently being edited
+
+            };
+
             $scope.newCommodityItem = function() {
                 $scope.newCommodity = !$scope.newCommodity;
                 $scope.commodity = {};
                 $scope.commodity.co_availability_date= moment().toDate();
                 $scope.commodity.co_until = moment().add(7, 'days').toDate();
             };
+
+            $scope.editCommodityItem = function(arrayindex) {
+
+                if (arrayindex == null) document.getElementById("commodities-table").style.visibility='visible'; //if cancel button was clicked
+
+                else document.getElementById("commodities-table").style.visibility='hidden';
+
+                $scope.editCommodity = !$scope.editCommodity;
+                $scope.commodity = $scope.commodities[arrayindex];
+                $scope.commodity.co_availability_date= moment().toDate();
+                $scope.commodity.co_until = moment().add(7, 'days').toDate();
+
+                $scope.editingCommodity = arrayindex;//sets the commodity currently being edited
+
+            };
+
+            $scope.updateCommodity = function() {
+
+                CommodityEditFactory.update({id:$scope.commodities[$scope.editingCommodity].fa_farmer._id, commodity_id:$scope.commodity._id}, $scope.commodity, function(success) {
+
+                    $window.scrollTo(0,0);
+
+                    showDialog($mdDialog, {statusText:"Successfully Updated!"}, false);
+
+                    $scope.editCommodity = false;
+
+                    document.getElementById("commodities-table").style.visibility='visible';
```

Comment:
@nick22891 whilst we are working with javascript, we should try our best to ensure we use angular's framework in dealing with element selection. In this case if you needed to set the visibility of an element, it could've been done using a flag with 'ng-show'. Will review the visuals and see whats the impact.